treeherder/seta/common.py

```diff
@@ -23,3 +29,54 @@ def job_priority_index(job_priorities):
         jp_index[key] = {'pk': jp.id, 'build_system_type': jp.buildsystem}
 
     return jp_index
+
+
+# The order of this is list is important as the more specific patterns
+# will be processed before the less specific ones. This must be kept up
+# to date with SETA_SUPPORTED_TC_JOBTYPES in settings.py.
+RE_JOB_TYPE_NAMES = [
+    {'name': 'test',         'pattern': re.compile('test-[^/]+/[^-]+-(.*)$')},
+    {'name': 'desktop-test', 'pattern': re.compile('desktop-test-[^/]+/[^-]+-(.*)$')},
+    {'name': 'android-test', 'pattern': re.compile('android-test-[^/]+/[^-]+-(.*)$')},
+    {'name': 'source-test',  'pattern': re.compile('(source-test-[^/]+)(?:/.*)?$')},
+    {'name': 'build',        'pattern': re.compile('(build-[^/]+)/[^-]+$')},
+    {'name': 'spidermonkey', 'pattern': re.compile('(spidermonkey-[^/]+)/[^-]+$')},
+    {'name': 'iris',         'pattern': re.compile('(iris-[^/]+)/[^-]+$')},
```

Comment:
build/spidermonkey/iris - these do not have a group to match on, I assume this will return the job name?

Reply:
The group is the stuff before the /.